jkube-kit/helm/src/main/java/org/eclipse/jkube/kit/resource/helm/HelmService.java

```diff
@@ -85,11 +92,17 @@ public class HelmService {
   private final JKubeConfiguration jKubeConfiguration;
   private final ResourceServiceConfig resourceServiceConfig;
   private final KitLogger logger;
+  private KubernetesClient kubernetesClient;
 
   public HelmService(JKubeConfiguration jKubeConfiguration, ResourceServiceConfig resourceServiceConfig, KitLogger logger) {
+    this(jKubeConfiguration, resourceServiceConfig, logger, null);
+  }
+
+  HelmService(JKubeConfiguration jKubeConfiguration, ResourceServiceConfig resourceServiceConfig, KitLogger logger, KubernetesClient kubernetesClient) {
```

Comment:
I'm reviewing and thinking about this PR since yesterday.

The fact that we're adding extra fields to this constructor doesn't feel right (especially one that's not a configuration but a heavy KubernetesClient instance).

We should probably be able to infer the configuration using `ClusterConfiguration.from(access, System.getProperties(), project.getProperties()).build();`

just like we do here:

https://github.com/eclipse-jkube/jkube/blob/ad4233e5b4fc0d6455481fec77ccfd7b510a8e05/jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/JKubeServiceHub.java#L128-L137

And then use the ClusterAccess#getConfig to initialize the temporary `.kube/config`

Maybe we can discuss this later, to see if there are better alternatives.
But storing an KubernetesClient instance here, doesn't seem right at all.

Reply:
I had added this mainly for testing. Exposing KubernetesClient field from constructor was causing other helm features to depend on K8s. It's only package private constructor, this KubernetesClient field would not be exposed via public constructor (it's passed as `null`)

Reply:
It doesn't matter, it's still not right to do so.
The helm module has no beef with the Kubernetes Client, it only has the dependency because the kubernetes-client dependency is high up in the hierarchy.
Let's say we want to split this in the future to allow for a helm-maven-plugin, the only dependency we'd probably want at that point would be the helm-java one, nothing else.